packages/server/src/fhir/repo.ts

```diff
@@ -1688,41 +1697,52 @@ export class Repository extends FhirRepository<PoolClient> implements Disposable
    * @param create - Flag for when "creating" vs "updating".
    * @returns The account value.
    */
-  private async getAccount(
+  private async getAccounts(
     existing: Resource | undefined,
     updated: Resource,
     create: boolean
-  ): Promise<Reference | undefined> {
-    const account = updated.meta?.account;
-    if (account && this.canWriteAccount()) {
+  ): Promise<Reference[] | undefined> {
+    const updatedAccounts = this.extractAccountReferences(updated.meta);
+    if (updatedAccounts && this.canWriteAccount()) {
       // If the user specifies an account, allow it if they have permission.
-      return account;
+      return updatedAccounts;
     }
 
     if (create && this.context.accessPolicy?.compartment) {
       // If the user access policy specifies a compartment, then use it as the account.
-      return this.context.accessPolicy.compartment;
+      return [this.context.accessPolicy.compartment];
```

Comment:
This leads to an interesting question about whether `accessPolicy.compartment` should also be an array.

Deferring that to future discussions about `AccessPolicy` v2